src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/ModelMetadata.cs

```diff
@@ -48,6 +48,12 @@ public class ModelMetadata
         }
 
         /// <summary>
+        /// Gets a collection of additional information about the model.
+        /// </summary>
+        public virtual IDictionary<string, object> AdditionalValues { get; }
```

Comment:
@rynowak @yishaigalatzer we agreed to make the change Ryan suggested below earlier today.  however the dictionary is case-insensitive in my current PR and in MVC 5.  switching to a case-sensitive `<object, object>` dictionary would probably affect very people.  but does this difference push either of you toward sticking with the current `<string, object>`, case-insensitive dictionary?